OpenRA.Mods.Common/AI/BaseBuilder.cs

```diff
@@ -37,10 +40,19 @@ public BaseBuilder(HackyAI ai, string category, Player p, PowerManager pm, Playe
 			playerPower = pm;
 			playerResources = pr;
 			this.category = category;
+			waterAvailable = true;
+			failCount = 0;
+			failRetryTicks = ai.Info.StructureProductionResumeDelay;
 		}
 
 		public void Tick()
 		{
+			// If failed to place something N consecutive times, wait N ticks until resuming building production
```

Comment:
Nit: that second N should be an M (it is a different arbitrary number to the first one).